controllers/bootstrap/authentication_bootstrap_controller.go

```diff
@@ -0,0 +1,172 @@
+
+func (r *BootstrapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, err error) {
+	log := logf.FromContext(ctx).WithName("controller_authentication_bootstrap")
+	subCtx := logf.IntoContext(ctx, log)
+	if subResult, err := r.makeAuthenticationCorrections(subCtx, req); subreconciler.ShouldHaltOrRequeue(subResult, err) {
+		if err != nil {
+			log.Error(err, "An error was encountered during Authentication bootstrap")
+		} else {
+			log.Info("Needed to requeue during Authentication bootstrap")
+		}
+		return subreconciler.Evaluate(subResult, err)
+	}
+	log.Info("Authentication bootstrap successful")
+	return
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *BootstrapReconciler) SetupWithManager(mgr ctrl.Manager) error {
+
+	authCtrl := ctrl.NewControllerManagedBy(mgr)
+
+	bootstrapPred := predicate.NewPredicateFuncs(func(o client.Object) bool {
+		return o.GetLabels()[ctrlcommon.ManagerVersionLabel] != version.Version
+	})
+
+	authCtrl.Watches(&operatorv1alpha1.Authentication{}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(bootstrapPred))
+	return authCtrl.Named("controller_authentication_bootstrap").Complete(r)
+}
+
+// makeAuthenticationCorrections handles changes that need to happen to the Authentication CR for compatibility reasons.
+func (r *BootstrapReconciler) makeAuthenticationCorrections(ctx context.Context, req ctrl.Request) (result *ctrl.Result, err error) {
+	log := logf.FromContext(ctx)
+	authCR := &operatorv1alpha1.Authentication{}
+	if result, err = r.getLatestAuthentication(ctx, req, authCR); subreconciler.ShouldHaltOrRequeue(result, err) {
+		return
+	}
+	if authCR.Labels[ctrlcommon.ManagerVersionLabel] == version.Version {
+		return subreconciler.ContinueReconciling()
+	}
+
+	if needsAuditServiceDummyDataReset(authCR) {
+		authCR.SetRequiredDummyData()
+	}
+
+	if err = r.writeConfigurationsToAuthenticationCR(ctx, authCR); err != nil {
+		log.Error(err, "Failed to update the Authentication")
+		return subreconciler.RequeueWithError(err)
+	}
+
+	authCR.Labels[ctrlcommon.ManagerVersionLabel] = version.Version
+
+	err = r.Update(ctx, authCR)
+	if err != nil {
+		log.Error(err, "Failed to update the Authentication")
+		return subreconciler.RequeueWithError(err)
+	}
+	log.Info("Performed bootstrap update to Authentication successfully")
+
+	return subreconciler.ContinueReconciling()
+}
+
+// writeConfigurationsToAuthenticationCR copies values from the
+// platform-auth-idp ConfigMap to the Authentication CR.
+func (r *BootstrapReconciler) writeConfigurationsToAuthenticationCR(ctx context.Context, authCR *operatorv1alpha1.Authentication) (err error) {
+	platformAuthIDPCM := &corev1.ConfigMap{}
+	if err = r.Get(ctx, types.NamespacedName{Name: "platform-auth-idp", Namespace: authCR.Namespace}, platformAuthIDPCM); k8sErrors.IsNotFound(err) {
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("failed to get ConfigMap: %w", err)
+	}
+	keys := map[string]any{
```

Comment:
Are these only keys which can be customisable ?

Reply:
Good point - I've added the missing keys to both this list as well as to the list that updatesAlways runs on.